src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -191,42 +135,79 @@ def export_deployment_as_template(resource_group_name, deployment_name):
     result = smc.deployments.export_template(resource_group_name, deployment_name)
     print(json.dumps(result.template, indent=2))#pylint: disable=no-member
 
-def tag_resource(
-        resource_group_name, resource_name, resource_type, tags, parent_resource_path=None,
-        api_version=None, resource_provider_namespace=None):
+def show_resource(resource_group_name=None, resource_provider_namespace=None,
+                  parent_resource_path=None, resource_type=None, resource_name=None,
+                  resource_id=None, api_version=None):
+    res = _ResourceUtils(resource_group_name, resource_provider_namespace,
+                         parent_resource_path, resource_type, resource_name,
+                         resource_id, api_version)
+    return res.get_resource()
+
+def delete_resource(resource_group_name=None, resource_provider_namespace=None,
+                    parent_resource_path=None, resource_type=None, resource_name=None,
+                    resource_id=None, api_version=None):
+    res = _ResourceUtils(resource_group_name, resource_provider_namespace,
+                         parent_resource_path, resource_type, resource_name,
+                         resource_id, api_version)
+    res.delete()
+
+def tag_resource(tags, resource_group_name=None, resource_provider_namespace=None,
+                 parent_resource_path=None, resource_type=None, resource_name=None,
+                 resource_id=None, api_version=None):
     ''' Updates the tags on an existing resource. To clear tags, specify the --tag option
     without anything else. '''
+    res = _ResourceUtils(resource_group_name, resource_provider_namespace,
+                         parent_resource_path, resource_type, resource_name,
+                         resource_id, api_version)
+    res.tag(tags)
+
+def list_resources(resource_group_name=None, resource_provider_namespace=None,
+                   resource_type=None, name=None, tag=None, location=None):
     rcf = _resource_client_factory()
-    resource = rcf.resources.get(
-        resource_group_name,
-        resource_provider_namespace,
-        parent_resource_path,
-        resource_type,
-        resource_name,
-        api_version)
-    # pylint: disable=no-member
-    parameters = GenericResource(
-        location=resource.location,
-        tags=tags,
-        plan=resource.plan,
-        properties=resource.properties,
-        kind=resource.kind,
-        managed_by=resource.managed_by,
-        sku=resource.sku,
-        identity=resource.identity)
-    try:
-        rcf.resources.create_or_update(
-            resource_group_name,
-            resource_provider_namespace,
-            parent_resource_path,
-            resource_type,
-            resource_name,
-            api_version,
-            parameters)
-    except CloudError as ex:
-        # TODO: Remove workaround once Swagger and SDK fix is implemented (#120123723)
```

Comment:
Yep, this is the existing code, but agree we can remove now 
[the related issue](https://github.com/Azure/azure-rest-api-specs/issues/308) is still active, but current [generated code](https://github.com/Azure/azure-sdk-for-python/blob/master/azure-mgmt-resource/azure/mgmt/resource/resources/operations/resources_operations.py#L432) already handles 202 , I think we are good. The new tests I added also worked. So we can close that issue as well